dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -849,13 +854,16 @@ private void checkCanShareWorkflow(User user, Workflow workflow) {
     @Timed
     @UnitOfWork
     @Path("/path/workfow/{repository}/permissions")
-    @ApiOperation(value = "Set the specified permission for a user on a workflow", authorizations = { @Authorization(value =  JWT_SECURITY_DEFINITION_NAME)}, notes = "Adds a permission for a workflow. The user must be the workflow owner.", response = Permission.class, responseContainer = "List")
+    @ApiOperation(value = "Set the specified permission for a user on a workflow", authorizations = { @Authorization(value =  JWT_SECURITY_DEFINITION_NAME)}, notes = "Adds a permission for a workflow. The user must be the workflow owner. Currently only supported on hosted workflows.", response = Permission.class, responseContainer = "List")
     public List<Permission> addWorkflowPermission(@ApiParam(hidden = true) @Auth User user,
             @ApiParam(value = "repository path", required = true) @PathParam("repository") String path,
             @ApiParam(value = "user permission", required = true) Permission permission) {
         Workflow workflow = workflowDAO.findByPath(path, false);
         checkEntry(workflow);
-        this.permissionsInterface.setPermission(workflow, user, permission);
+        if (workflow.getMode() != WorkflowMode.HOSTED) {
```

Comment:
Maybe add a comment `TODO` with issue number for an issue relaxing this. 
(i.e. make it easier for the next developer to search for the places they'll need to change)

Reply:
👍